internal/controller/instaslice_daemonset.go

```diff
@@ -1129,3 +1014,84 @@ func walkMigDevices(d nvml.Device, f func(i int, d nvml.Device) error) error {
 	}
 	return nil
 }
+
+func populateMigDeviceInfos(device nvml.Device) (map[string]*MigDeviceInfo, error) {
```

Comment:
would it be possible to write a unit test for this function?

Reply:
I also think maybe we can move this from this file as it does not actually depend on controller logic.

Since this is a pure function, unit testing should be simple and we can test it without needed to mock the KubeClient..

Reply:
Thanks, yes working on it.

Reply:
This method takes input as the vendor device, which makes it necessary to implement around 50+ methods around NVML interface, I tried to mock around 10 and then stopped, unit testing is cumbersome until we get a mocking framework or codegen.